go/vt/vtgate/planbuilder/operators/route_planning.go

```diff
@@ -333,6 +333,57 @@ func tryRewriteOrToIn(expr sqlparser.Expr) sqlparser.Expr {
 	return nil
 }
 
+func tryRewriteOrFalse(expr sqlparser.Expr) sqlparser.Expr {
+	rewrote := false
+	newPred := sqlparser.Rewrite(sqlparser.CloneExpr(expr), func(cursor *sqlparser.Cursor) bool {
+		_, ok := cursor.Node().(*sqlparser.OrExpr)
+		return ok
+	}, func(cursor *sqlparser.Cursor) bool {
+		// we are looking for the pattern WHERE c = 1 OR 1 = 0
+		switch or := cursor.Node().(type) {
+		case *sqlparser.OrExpr:
+			isFalse := func(subExpr sqlparser.Expr) bool {
+				cmp, ok := subExpr.(*sqlparser.ComparisonExpr)
+				if !ok {
+					return false
+				}
+
+				left, ok := cmp.Left.(*sqlparser.Literal)
+				if !ok {
+					return false
+				}
+
+				right, ok := cmp.Right.(*sqlparser.Literal)
+				if !ok {
+					return false
+				}
+
+				switch cmp.Operator {
+				case sqlparser.EqualOp:
+					if left.Val != right.Val {
+						return true
+					}
+				}
```

Comment:
I think perhaps we should use the evalengine's Evaluate method to evaluate more comparisons. That way we would be able to support more than just equalities.

WDYT @systay

Reply:
@frouioui yes I was hoping someone would say that! Can you point me to where this method is defined?

Reply:
For sure. First you'll want to use `evalengine.Translate` to translate the `sqlparser.Expr` into an evalengine expression. Then, you'll need to create and envalengine environment using `evalengine.EmptyExpressionEnv()`, and finally call the `Evaluate` method on your environment with the comparison expression.

It gives us the following snippet:

```go
func TestEvalEngineCmp(t *testing.T) {
	predicate := &sqlparser.ComparisonExpr{
		Operator: sqlparser.EqualOp,
		Left:     sqlparser.NewIntLiteral("1"),
		Right:    sqlparser.NewIntLiteral("1"),
	}

	pred, err := evalengine.Translate(predicate, &dummyTranslator{})
	require.NoError(t, err)

	env := evalengine.EmptyExpressionEnv()
	res, err := env.Evaluate(pred)
	require.NoError(t, err)

	t.Log(res.String()) // will print "INT64(1)"
}
```

Note that you will need to `Translate` the predicate using the semantic table that is available in the planning context.

Reply:
Okay I found `Evaluate()` but I'm not sure how I can use it with a `sqlparser.Expr`... trying to find out how ~ help is welcome. 🙂

Reply:
@pudiva, can you see this comment: https://github.com/vitessio/vitess/pull/11653#discussion_r1015720711? I am happy to explain again otherwise 😃

Reply:
sorry your comments didn't show up here! 👀

Reply:
No worries! Let me know if I am missing something in the original response for your problem